internal/impl/snowflake/streaming/userdata_converter.go

```diff
@@ -252,26 +251,9 @@ func (c binaryConverter) ValidateAndConvert(stats *statsBuffer, val any, buf typ
 		buf.WriteNull()
 		return nil
 	}
-	var v []byte
-	switch t := val.(type) {
-	case string:
-		if t != "" {
-			// We don't modify this byte slice at all, so this is safe to grab the bytes
-			// without making a copy.
-			// Also make sure this isn't an empty string because it's undefined what the
-			// value is.
-			v = unsafe.Slice(unsafe.StringData(t), len(t))
-		} else {
-			v = []byte{}
-		}
-	case []byte:
-		v = t
-	default:
-		b, err := bloblang.ValueAsBytes(val)
-		if err != nil {
-			return err
-		}
-		v = b
```

Comment:
Nice cleanup! Was this added initially due to some perf issues in the bloblang code?

Reply:
I was over optimizing the string conversion. I don't think it's worth the risk now